src/jit/importer.cpp

```diff
@@ -4580,16 +4594,25 @@ void           Compiler::impImportAndPushBox (CORINFO_RESOLVED_TOKEN * pResolved
         impBoxTempInUse = true;
             
 #ifdef FEATURE_READYTORUN_COMPILER
-        if (opts.IsReadyToRun())
+        if (opts.IsReadyToRun() && !fNeedsRuntimeLookup)
         {
             op1 = impReadyToRunHelperToTree(pResolvedToken, CORINFO_HELP_READYTORUN_NEW, TYP_REF);
         }
         else
 #endif
         {
-            op1 = gtNewHelperCallNode(  info.compCompHnd->getNewHelper(pResolvedToken, info.compMethodHnd),
-                                        TYP_REF, 0,
-                                        gtNewArgList(op2));
+            // TODO: ReadyToRun: When generic dictionary lookups are necessary, replace the lookup call
+            // and the newfast call with a single call to a dynamic R2R cell that will:
+            //      1) Load the context
+            //      2) Perform the generic dictionary lookup and caching, and generate the appropriate stub
+            //      3) Allocate and return the new object for boxing
+            // Reason: performance (today, we'll always use the slow helper for the R2R generics case)
+
+            CorInfoHelpFunc helper = opts.IsReadyToRun() ?
+                CORINFO_HELP_NEWFAST :
+                info.compCompHnd->getNewHelper(pResolvedToken, info.compMethodHnd);
```

Comment:
Yes, this sounds like a much better idea!

---
In reply to: [62789829](https://github.com/dotnet/coreclr/pull/4519#discussion_r62789829) [](ancestors = 62789829)